devices/oneplus-enchilada/default.nix

```diff
@@ -0,0 +1,126 @@
+{ config, lib, pkgs, ... }:
+
+{
+  mobile.device.name = "oneplus-enchilada";
+  mobile.device.identity = {
+    name = "OnePlus 6";
+    manufacturer = "OnePlus";
+  };
+
+  mobile.hardware = {
+    soc = "qualcomm-sdm845";
+    ram = 1024 * 8;
+    screen = {
+      width = 1080; height = 2280;
+    };
+  };
+
+  mobile.boot.stage-1 = {
+    compression = "xz";
+    kernel.package = pkgs.callPackage ./kernel { };
+    firmware = [
+      config.mobile.device.firmware
+    ];
+  };
+
+
+  mobile.device.firmware = pkgs.callPackage ./firmware {};
+
+  mobile.system.android.device_name = "OnePlus6";
```

Comment:
just out of interest, is there any way way to have a oneplus-common definition and derive enchilada and fajita from it? we have this problem in postmarketOS too where we have to duplicate all of the common properties.

Reply:
There absolutely is. In fact there's several ways to do it. That's the problem. How does mobile NixOS want to support "families"? At this point, I just don't want it to be the blocker for enchilada support.

Reply:
@MilesBreslin got it: instead of creating the wrong abstraction, I prefer erring on the side of cautiousness and have multiple instances of the same problem to start solving the problem.